### PR TITLE
Added ipad/left click support

### PR DESCRIPTION
**Pull request by glaidler**
Minor alteration to allow left-click and ipad support.

### CODE REVIEW

File: contextMenu.js [hunk 1]
```diff
@@ -275,20 +275,36 @@ angular.module('ui.bootstrap.contextMenu', [])
...
+        var openMenus = function(event) {
+
             event.stopPropagation();
-            $scope.$apply(function () {
+            $scope.$apply(function() {
                 event.preventDefault();
                 var options = $scope.$eval(attrs.contextMenu);
                 var model = $scope.$eval(attrs.model);
                 if (options instanceof Array) {
-                    if (options.length === 0) { return; }
```
Comment (josetaira) [inline contextMenu.js]: Unnecessary format. Please revert

File: contextMenu.js [hunk 2]
```diff
@@ -275,20 +275,36 @@ angular.module('ui.bootstrap.contextMenu', [])
 
         contextMenus.push($ul);
     };
-    return function ($scope, element, attrs) {
-        element.on('contextmenu', function (event) {
+    return function($scope, element, attrs) {
+        var openMenus = function(event) {
+
             event.stopPropagation();
-            $scope.$apply(function () {
```
Comment (josetaira) [inline contextMenu.js]: Unnecessary reformat, please revert this line

File: contextMenu.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ angular.module('ui.bootstrap.contextMenu', [])
 })
 .directive('contextMenu', ["$parse", "$q", "CustomService", "$sce", function ($parse, $q, custom, $sce) {
 
-    var contextMenus = [];
+     var contextMenus = [];
```
Comment (josetaira) [inline contextMenu.js]: Unnecessary reformat. please revert this line

**Comment by josetaira** [comment]
@glaidler changes look good. Can you check the notes I put and revert the unnecessary format changes before we merge this in?

**Comment by glaidler** [comment]
No problem will do.  Was my ide being "helpful".

G



*http://accelerata.co.uk* <http://accelerata.co.uk>
tel: 01392 580154
mob: 07968 472239
skype: glaidler

**Comment by glaidler** [comment]
Hi - formatting undone.

File: contextMenu.js [hunk 4]
```diff
@@ -289,6 +290,19 @@ angular.module('ui.bootstrap.contextMenu', [])
                     throw '"' + attrs.contextMenu + '" not an array';
                 }
             });
-        });
-    };
+
+
+        };
+
+        if (attrs.hasOwnProperty('contextLeftClick')) {
```
Comment (josetaira) [inline contextMenu.js]: The registration of click from here to a few lines below makes the context menu trigger twice when doing a right click on web.

File: contextMenu.js [hunk 5]
```diff
@@ -289,6 +290,19 @@ angular.module('ui.bootstrap.contextMenu', [])
...
+            element.on('click ', openMenus);
+            element.on('click ', function(e) {
+                if ($(e.target).hasClass('dropdown')) {
```
Comment (josetaira) [inline contextMenu.js]: Error here: `ReferenceError: $ is not defined`

**Comment by josetaira** [comment]
Found some issues while testing this out. `ReferenceError: $ is not defined` seems to be problematic not just in this case. I'll push a fix for this issue later, but for now the more pressing issue is the click.

I think in general, we should be able to specify custom events that openMenus triggers to as requested in https://github.com/Templarian/ui.bootstrap.contextMenu/issues/43

Seems like it'd be easy to update the implementation to be more customizable.

Property name should be `context-menu-custom-trigger`

**Comment by glaidler** [comment]
Yes - this makes total sense i will adjust.

G



*http://accelerata.co.uk* <http://accelerata.co.uk>
tel: 01392 580154
mob: 07968 472239
skype: glaidler

**Comment by josetaira** [comment]
Already fixed in https://github.com/Templarian/ui.bootstrap.contextMenu/pull/66

The PR seems to have become unnecessary.

**Pull request closed without merging**